sdk/core/azure-core-amqp/inc/azure/core/amqp/network/transport.hpp

```diff
@@ -16,60 +17,119 @@ namespace Azure { namespace Core { namespace Amqp { namespace Network { namespac
 }}}}} // namespace Azure::Core::Amqp::Network::_detail
 
 namespace Azure { namespace Core { namespace Amqp { namespace Network { namespace _internal {
-  enum class TransportState
-  {
-    Closed,
-    Closing,
-    Open,
-    Opening,
-    Unknown
-  };
 
-  enum class TransportOpenResult
+  /** @brief Status of a transport open operation.
+   *
+   *
+   */
+  enum class TransportOpenStatus
   {
+    /// Transport is in invalid state.
     Invalid,
+    /// Transport open successful.
     Ok,
+    /// Transport open failed with an error.
     Error,
+    /// Transport open was cancelled.
     Cancelled
   };
-  enum class TransportSendResult
+
+  /** @brief Status of a transport send operation.
+   *
+   */
+  enum class TransportSendStatus
   {
+    /// Send status Invalid - this should never be returned.
     Invalid,
+
+    /// Send status unknown
     Unknown,
+
+    /// Send successful
     Ok,
+
+    /// Send completed with an error.
     Error,
+
+    /// Send was cancelled.
     Cancelled
   };
 
   class Transport;
 
+  /** Transport operational events.
+   *
+   * Normally a client will not have to register for these events, but they are available for
+   * clients if necessary.
+   */
   struct TransportEvents
   {
-    virtual void OnOpenComplete(TransportOpenResult openResult) = 0;
+    /** @brief Called when bytes are received on the transport.
+     *
+     * @param transport The transport on which bytes are received.
+     * @param buffer The buffer containing the bytes received.
+     * @param size The size of the buffer.
+     *
+     */
     virtual void OnBytesReceived(
         Transport const& transport,
         const unsigned char* buffer,
         size_t size)
         = 0;
+
+    /** @brief Called when an error has occurred on the transport */
     virtual void OnIoError() = 0;
```

Comment:
```suggestion
    virtual void OnIOError() = 0;
```

Two-letter abbreviations don't get pascalized.